master/internal/api_workspace.go

```diff
@@ -423,13 +423,51 @@ func (a *apiServer) PatchWorkspace(
 		insertColumns = append(insertColumns, "uid", "user_", "gid", "group_")
 	}
 
-	if req.Workspace.DefaultComputePool != "" {
-		updatedWorkspace.DefaultComputePool = req.Workspace.DefaultComputePool
-		insertColumns = append(insertColumns, "default_compute_pool")
-	}
-	if req.Workspace.DefaultAuxPool != "" {
-		updatedWorkspace.DefaultAuxPool = req.Workspace.DefaultAuxPool
-		insertColumns = append(insertColumns, "default_aux_pool")
+	if req.Workspace.DefaultAuxPool != "" || req.Workspace.DefaultComputePool != "" {
+		if err = workspace.AuthZProvider.Get().
+			CanSetWorkspacesDefaultPools(ctx, currUser, currWorkspace); err != nil {
+			return nil, status.Error(codes.PermissionDenied, err.Error())
+		}
+
+		rpConfigs, err := a.resourcePoolsAsConfigs()
+		if err != nil {
+			return nil, err
+		}
+		rpNames, _, err := db.ReadRPsAvailableToWorkspace(
+			ctx, currWorkspace.Id, 0, -1, rpConfigs,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if req.Workspace.DefaultComputePool != "" {
+			found := false
+			for _, rpName := range rpNames {
+				if rpName == req.Workspace.DefaultComputePool {
+					found = true
+				}
+			}
```

Comment:
Nit: you could just make a `Set` out of the slice of strings using [`FromSlice`](https://pkg.go.dev/github.com/determined-ai/determined/master@v0.0.0-20230811011746-bbe70e07cbe4/pkg/set#FromSlice) and then skip these loops. It'd be less code and probably faster—definitely better time complexity.

```go
tmp, _, err := db.ReadRPsAvailableToWorkspace(
	ctx, currWorkspace.Id, 0, -1, rpConfigs,
)
if err != nil {
	return nil, err
}
rpNames := set.FromSlice(tmp)

if req.Workspace.DefaultComputePool != "" {
	if !rpNames.Contains() {
		return nil, status.Error(codes.FailedPrecondition, "unable to bind a resource "+
			"pool that does not exist or is not available to the workspace")
	}
	updatedWorkspace.DefaultComputePool = req.Workspace.DefaultComputePool
	insertColumns = append(insertColumns, "default_compute_pool")
}
```